docs/design/proposals/codegen-refactor/centralize_1.md

```diff
@@ -0,0 +1,197 @@
+# Centralize config access and reconciliation
+
+### Key Terms
+* `ackgenconfig`: the [code](https://github.com/aws-controllers-k8s/code-generator/blob/main/pkg/generate/config/config.go#L24) representation of *generator.yaml*. an **input** to *code-generator*.
+* `resource` | `k8s-resource` | `ackcrd`: the [code](https://github.com/aws-controllers-k8s/code-generator/blob/main/pkg/model/crd.go#L63) reprensenting a single top-level resource in an AWS Service. *code-generator* generates these resources using heuristics and `ackgenconfig`.
+* `shape` | `aws-sdk` | `sdk-shape` | `sdk`: the original operations, models, errors, structs for a given AWS service. sourced from *aws-sdk*, ex: [aws-sdk-go s3](https://github.com/aws/aws-sdk-go/blob/main/models/apis/s3/2006-03-01/api-2.json#L1)
+* `reconciliation`: logic involving access to or relations between `resource`, `shape`, `ackgenconfig`, and `aws-sdk`
+* `ackmodel`: the [code](https://github.com/aws-controllers-k8s/code-generator/blob/main/pkg/model/model.go#L36) representation of ACK's view of the world; the source of truth for `aws-sdk`, `ackgenconfig`, and `reconciliation`
+
+## Problem
+It is becoming increasingly difficult to contribute to and maintain *code-generator* due to unclear encapsulations and dispersed logic throughout the codebase. For example, `ackgenconfig` is passed throughout code and accessed in a variety of ways, and logic involving `reconciliation` is inconsistent, both in implementation and interface. Some specific examples include:
+  * [overextended encapsulations](#overextended-encapsulations)
+  * [long function parameter lists](#long-parameters)
+  * [inconsistent use of `ackgenconfig`](#ackgenconfig-use)
+  * [increased cognitive load for contributors](#more-work-for-contributors)
+
```

Comment:
Might I add, while we are going for this, inconsistent use of `resource` structs i.e. references to `latest` and `desired`, and setting properties on either seemingly arbitrarily at points.

Reply:
resolving; this was related to `runtime`